src/core/runner.ts

```diff
@@ -410,6 +407,15 @@ export default class Runner {
       journey.callback({ params: options.params } as any);
       journey.monitor.update(this.monitor?.config);
       journey.monitor.validate();
+
+      /**
+       * Add the tags created through `monitor.use` to the journey
+       */
+      journey.addTags(journey.monitor.config.tags);
+      if (!journey.isMatch(options.match, options.tags)) {
```

Comment:
I don't think this is the correct behaviour. Monitor.use should never influenced the local runner. It should be only applicable to the pushed monitors.